main/network.c

```diff
@@ -200,6 +200,10 @@ PHPAPI int php_network_getaddresses(const char *host, int socktype, struct socka
 
 	if ((n = getaddrinfo(host, NULL, &hints, &res))) {
 		if (error_string) {
+			/* free error string received during previous iteration (if any) */
+			if (*error_string) {
+				zend_string_release_ex(*error_string, 0);
+			}
```

Comment:
Is it feasible to push this up into the caller? It seems somewhat odd to deal with it here.

Reply:
These changes are all in `php_network_getaddresses()` which is exported. We could fix all calls from core and bundled extensions, but external extensions may have the same memory leak.

Also note that I copied that snippet from `php_network_connect_socket_to_host()`, and that had been introduced with 79d649e17323f4fa6bd234f4cdf609a99e91435a (which fixed a similar memory leak long ago).

Maybe leave it this way for PHP 7, and push up into the caller for PHP 8?